### ISSUE TITLE
Accessibility: WAI-ARIA roles nesting wrong in menu

### ISSUE THREAD

**Issue by plastikschnitzer** [magento/magento2#35889]
### Preconditions and environment

- Magento 2.4.4
- Blank theme, Luma theme

### Steps to reproduce

Navigate to storeview
analyze html code of menu for correct ARIA markup

### Expected result

Menu should be structured according to ARIA schema the following way:
```
<ul role="menu">
 <li role="none">
  <a href="" role="menuitem">link</a>
 </li>
</ul>
```
### Actual result

```
<ul role="menu">
 <li>
  <a href="" role="menuitem">link</a>
 </li>
</ul>
```

As the `li` tag has no role, the child element for the menu role is missing and the parent element for the menuitem role is missing. 

Users who rely on accessibility tools might run into a deadend here.

### Additional information

This is also a ranking factor for search engines and part of the lighthouse check, should be fixed!

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [x] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by plastikschnitzer** [comment]
@magento give me 2.4-develop instance

**Comment by plastikschnitzer** [comment]
Yes, I can confirm the issue can be reproduced on Magento-2.4-develop

**Comment by engcom-November** [comment]
Hi @plastikschnitzer ,
Verified this on 2.4-develop instance but could not find the role: menu in "Accessibility Properties". Can you please elaborate which particular page and elements you are referring to check the issue.
Thank you.

**Comment by engcom-November** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-November** [comment]
Hi @plastikschnitzer ,
Verified the behavior on Magento 2.4-develop branch and is reproducible. But however, Accessibility: WAI-ARIA  is not mentioned in Magento [docs](https://developer.adobe.com/commerce/php/coding-standards/html-style-guide/) instead mentioned as Web Content Accessibility Guidelines (WCAG) 2.0. Kindly refer the document and provide supporting docs/proofs to consider this as a valid issue.
Thank you.

**Comment by engcom-November** [comment]
Hi @plastikschnitzer , Thank you for reporting and collaboration.
Verified the behavior on Magento 2.4-develop branch and is reproducible. But however, Accessibility: WAI-ARIA is not mentioned anywhere in Magento [docs](https://developer.adobe.com/commerce/php/coding-standards/html-style-guide/) instead mentioned as Web Content Accessibility Guidelines (WCAG) 2.0. Kindly refer the document and provide supporting docs/proofs that support ARIA compliance for Magento pages.
Thank you.

**Comment by plastikschnitzer** [comment]
Thank you @engcom-November thank you for confirming and I am happy to precise further:
I was stumbling over this issue running a default google lighthouse test on the shop: https://web.dev/measure
Accessibility and UX is becoming more and more a ranking factor in search engines, Magento provides the role element already, but it needs a slight modification as stated in my opening posting with the code proposal, it should work then as expected by WAI-ARIA and search engines will be happy ;-) 
Further details can be found on the WAI-ARIA documentation: https://www.w3.org/TR/wai-aria-1.1/#menu and https://www.w3.org/TR/wai-aria-1.1/#menuitem

As the nesting in Magento has the `<li>` element without any role in between the `<ul role="menu">` and `<a href="" role="menuitem">`, the WAI-ARIA compatibility check fails as it expects a role on the `<li>`. `<li role="none">`  or <li role="presentation"> excludes the <li> from the nesting tree and the test will be passed.

Check https://www.w3.org/TR/wai-aria-1.1/#presentation and its examples.

**Comment by plastikschnitzer** [comment]
I have a quick fix which works for the product category menu, insert the code into
Content –> Configuration –> Edit on highest level –> Footer –>  Miscellaneous HTML

```
<script>
let mainnav = document.getElementById("store.menu");
let addmenurole = mainnav.getElementsByTagName("li");
for (let i = 0; i < addmenurole.length; i++) {
  addmenurole[i].setAttribute("role","none");}
</script>
```

This will add `role="none"` to the `li` of the main menu and fix the WAI-ARIA compatibility issues. A proper fix should be introduced in the same place where the other role attributes are set, this is just a quick temporary fix, also to demonstrate how the markup should look like.

**Comment by engcom-November** [comment]
Thank you for the response @plastikschnitzer . We are considering this as feature request but not a bug as WAI-ARIA compliance is not mentioned anywhere in Magento [docs](https://developer.adobe.com/commerce/php/coding-standards/html-style-guide/).

**Comment by loic-paquin** [comment]
This demo website is in 2.4.0 : [demo](https://magento-demo.mageplaza.com/)

As you can see the `<li>` in the menu have the attribute `role="presentation"`
Also my websites in 2.4.3 does not have the issue.
My websites in 2.4.4 have the issue.

It is not a feature request since it was working properly in the previous versions.

**Comment by mirianstangherlin** [comment]
@magento I am working on this

**Comment by engcom-Hotel** [comment]
Thank you @loic-paquin for your inputs [here](https://github.com/magento/magento2/issues/35889#issuecomment-1294354685). We are confirming this issue to process it further.

Thanks
